### ISSUE TITLE
invalid region is not caught

### ISSUE THREAD

**Issue by dschwoerer** [boutproject/BOUT-dev#1567]
I tried:
```
  REGION invalid = static_cast<REGION>(99999);

  // This is not a valid region for Field3D                                                                                                                                  
  EXPECT_THROW(field.getRegion(invalid), BoutException);
```
which does not throw a BoutException.

Running it normally produces the output:
```
terminate called after throwing an instance of 'std::out_of_range'
  what():  map::at
Aborted (core dumped)
```

I assume this is not what we should expect ...

**Comment by d7919** [comment]
Ah yes I think I see what's happening here --

~~~~
  const auto found = regionMap3D.find(region_name);
  if (found == end(regionMap3D)) {
    throw BoutException(_("Couldn't find region %s in regionMap3D"), region_name.c_str());
  }
  return found->second;
~~~~

is the code in `Mesh` -- we check if we can find the region and throw a `BoutException` if not. But here you're going through `Field` and passing the `enum` -- because the `Mesh` lookup is based on a `std::string` we attempt to convert this `enum` to `string` which involves a lookup in a map -- this is not guarded so ends up throwing from the `map` internals (as we use `std::map::at` to access, which throws if the key is not present).

The simple fix for this test is to use an invalid string in the argument ("RGN_THAT_IS_NOT_DEFINED").  In the long term I think the plan is to remove the `enum` use or convert to an `enum class` as these do not work with user registration.

**Comment by dschwoerer** [comment]
Fixed in 430b429
Should this go to master?

**Comment by ZedThree** [comment]
Fixed in #1570
